components/TxConfirmationDialog.qml

```diff
@@ -0,0 +1,465 @@
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; primary: false }
+            PropertyChanges { target: confirmButton; visible: true }
+        }, State {
+            // error message being displayed, show only back button
+            name: "error";
+            when: errorText.text !== ""
+            PropertyChanges { target: dialogTitle; text: "Error" }
+            PropertyChanges { target: errorText; visible: true }
+            PropertyChanges { target: txAmountText; visible: false }
+            PropertyChanges { target: txAmountBusyIndicator; visible: false }
+            PropertyChanges { target: txFiatAmountText; visible: false }
+            PropertyChanges { target: txDetails; visible: false }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: false }
+            PropertyChanges { target: buttons; visible: true }
+            PropertyChanges { target: backButton; visible: true; primary: true }
+            PropertyChanges { target: confirmButton; visible: false }
+        }, State {
+            // creating or sending transaction, show tx details and don't show any button
+            name: "bottomText";
+            when: errorText.text == "" && bottomText.text !== ""
+            PropertyChanges { target: errorText; visible: false }
+            PropertyChanges { target: txAmountText; visible: root.transactionAmount !== "(all)" || (root.transactionAmount === "(all)" && currentWallet.isHwBacked() === true) }
+            PropertyChanges { target: txAmountBusyIndicator; visible: !txAmountText.visible }
+            PropertyChanges { target: txFiatAmountText; visible: txAmountText.visible && persistentSettings.fiatPriceEnabled && root.transactionAmount !== "(all)" }
+            PropertyChanges { target: txDetails; visible: true }
+            PropertyChanges { target: bottom; visible: true }
+            PropertyChanges { target: bottomMessage; visible: true }
+            PropertyChanges { target: buttons; visible: false }
+        }
+    ]
+
+    // same signals as Dialog has
+    signal accepted()
+    signal rejected()
+
+    function open() {
+        root.visible = true;
+
+        //clean previous error message
+        errorText.text = "";
+        root.forceActiveFocus()
+    }
+
+    function close() {
+        root.visible = false;
+    }
+
+    function clearFields() {
+        root.transactionAmount = "";
+        root.transactionAddress = "";
+        root.transactionDescription = "";
+        root.transactionFee = "";
+        root.transactionPriority = "";
+        root.sweepUnmixable = false;
+    }
+
+    function showFiatConversion(valueXMR) {
+        return (fiatApiConvertToFiat(valueXMR) === "0.00" ? "<0.01 " + fiatApiCurrencySymbol() : "~" + fiatApiConvertToFiat(valueXMR) + " " + fiatApiCurrencySymbol());
```

Comment:
https://github.com/monero-project/monero-gui/blob/f5c5df7b677e10c78bddb2defa7b392bfe42696a/pages/Transfer.qml#L333

Reply:
I think this should be:

```
const fiatFee = fiatApiConvertToFiat(valueXMR);
return "%1 %2".arg(fiatFee < 0.01 ? "<0.01" : "~" + fiatFee).arg(fiatApiCurrencySymbol()); 
```

Right?
I don't know why you used %3 instead of %2.

Reply:
Right